sys/fmt/fmt.c

```diff
@@ -487,21 +486,23 @@ uint32_t scn_u32_hex(const char *str, size_t n)
     uint32_t res = 0;
 
     while (n--) {
-        char c = *str++;
-        if (!fmt_is_digit(c)) {
-            if (fmt_is_upper(c)) {
-                c = _to_lower(c);
-            }
-            if (c == '\0' || c > 'f') {
-                break;
-            }
-            res <<= 4;
-            res |= c - 'a' + 0xa;
+        unsigned c = *str++;
+        unsigned d;
+        if ( '0'<= c && c <= '9'){
+            d = c - '0';
+        }
+        else if ( 'A' <= c && c <= 'F' ) {
+            d = c - 'A' + 0xa;
+        }
+        else if ( 'a' <= c && c <= 'f' ) {
+            d = c - 'a' + 0xa;
```

Comment:
```suggestion
            d = c - (unsigned)'a' + 0xa;
```

Reply:
there shall never be an overflow (passed the condition) -> this is defined even if a is promoted to int.

Reply:
I could add it for peace of mind